### ISSUE TITLE
Scoped Accounts

### ISSUE THREAD

**Issue by treequell** [GrapheneOS/os-issue-tracker#2244]
Enhancement

Scope apps access to Accounts - for those with the `find accounts on the device` permission.

**Comment by treequell** [comment]
How - A Picker so that I can choose which Accounts an app has access to. In the same way that Contact and Storage Scopes work.

Why - Two reasons:
- The Accounts list contains PII. Some Accounts use my email address as their Name. This could be used by Apps which request this permission as an Identifier. 
- The ability to sign into my Google Account in one Google app only. Currently, when I sign into one Google app, I get signed in to all Google apps I have installed.

**Comment by treequell** [comment]
It's just an idea I had following conversations with a couple of community members.

I suppose it boils down to whether the Accounts list with my email addresses can be used as an Identifier by other apps.

**Comment by muhomorr** [comment]
`GET_ACCOUNTS` ("find accounts on the device") permission doesn't allow the app to get the full list of accounts, unless it's a legacy app that targets SDK lower than 26 (Android 8, released in 2017).

See https://developer.android.com/about/versions/oreo/android-8.0-changes#aaad

**Comment by treequell** [comment]
> `GET_ACCOUNTS` ("find accounts on the device") permission doesn't allow the app to get the full list of accounts, unless it's a legacy app that targets SDK lower than 26 (Android 8, released in 2017).
> 
> See https://developer.android.com/about/versions/oreo/android-8.0-changes#aaad

Oh that's great, I wasn't aware of that. Thank you for providing the link.

> * The ability to remain signed out of an App, even if I have an Account for that App in the list of Accounts. Example: Say I want to sign into my Google Account only on YouTube. Currently, I get signed in to all my other Google apps too (Maps, Play Store, etc.).

There is still this aspect. To avoid this issue, I have instructed community members to use separate user profiles.

**Comment by mtalexan** [comment]
How would this act with the sandboxed Google Apps? If I'm not mistaken, apps that need/use Google Services Framework effectively use the sandboxed gapps as an account proxy. The Gapps are the ones actually linked to a Google Account (or not), and the apps connect to GApps.  Correct?

**Comment by thestinger** [comment]
They are regular apps and can only do what regular apps can do. There's no special account functionality and it has no special relevance to this.

**Comment by mtalexan** [comment]
> They are regular apps and can only do what regular apps can do. There's no special account functionality and it has no special relevance to this.

Indeed, but few other apps are used by almost every other app as a proxy system for account authentication. That behavior is something all apps are capable of, but not many apps other than GApps do. I bring it up only because the specific example was about Google apps sharing account login, which might(?) be relying on GApps having Google Account access rather than directly checking for the Google account from the system accounts.

**Comment by xttlegendapi** [comment]
This should probably be a priority.

Example: If I now login to play services, directly my camera app is logged in (even without network permission) and other apps also use the google account to show popups if I want to sign in or remember passwords or something.

Doesn't feel private, probably isn't..

**Comment by thestinger** [comment]
Apps within a profile can communicate with mutual consent between them. This wouldn't change that. It doesn't really work the way you seem to think.

**Comment by xttlegendapi** [comment]
Sorry I understand it wrong then. What is/do you mean by mutual content. any sources?

**Comment by mtalexan** [comment]
> This should probably be a priority.
> 
> Example: If I now login to play services, directly my camera app is logged in (even without network permission) and other apps also use the google account to show popups if I want to sign in or remember passwords or something.
> 
> Doesn't feel private, probably isn't..

This is what I was talking about. The the Google Services effectively act as a proxy to accounts.  Instead of the GCam looking directly at accounts, almost every case of account usage actually uses Google Services and Google Services is what access the accounts.


> Apps within a profile can communicate with mutual consent between them. This wouldn't change that. It doesn't really work the way you seem to think.

The "mutual consent" there is doing a lot of heavy lifting. The consent is between the two apps, and there's virtually no ability for the user to block that consent. The most common form of mutual consent is between an app and the Google Services, which most app developers take as all but mandatory since it is virtually mandatory on a fully Googled-up phone. 

I think the intent of this Issue could really only be addressed by supporting toggles for a user to allow or disallow communication between apps that already have mutual consent. I don't know if that's possible since it would be like a per-app to per-app localhost firewall.

**Comment by treequell** [comment]
Scoping app communication is a separate issue, see: #2197

**Comment by mtalexan** [comment]
> Scoping app communication is a separate issue, see: #2197

Excellent.  
I was just saying that I think a lot of people aren't completely clear that the majority of their "app linked to account" cases are actually "app talks to Google Services" and "Google Services linked to account". I'm honestly not sure how many cases of direct "app linked to account" exist.  

In terms of priorities, this probably doesn't need to be very high given that the majority of use cases seem to be a misunderstanding of how accounts are accessed/used.

**Comment by xttlegendapi** [comment]
Sorry for commenting here without knowing what I'm talking about. Now it's clear for me I guess.

The account syncing is part of a mutual consent rule between play services and other google apps like google photos. Can this communication be used to avoid network or other sensor restrictions? In theory yes, right? 

In that case, #2197 should be a priority

**Comment by treequell** [comment]
> The account syncing is part of a mutual consent rule between play services and other google apps like google photos.

That's not how AOSP works.

As I stated when I opened this feature request, and clarified by @muhomorr, it's about scoping apps access to the GET_ACCOUNTS permission, which is already restricted anyway.

@xttlegendapi You seem to misunderstand how this permission works. We can move to the discussion forum https://discuss.grapheneos.org/, and I'll be happy to talk about this in more detail. This thread is not the appropriate place for this discussion.

**Comment by thestinger** [comment]
> The consent is between the two apps, and there's virtually no ability for the user to block that consent.

Users choose which apps are installed together in profiles.
